module/hooks/spell.mjs

```diff
@@ -0,0 +1,43 @@
+const HOOKS = {};
+
+/* -------------------------------------------- */
+
+HOOKS.protectiveMirage = {
+  defendAttack(item, action, _origin, rollData) {
+    if ( !action.tags.has("strike") ) return;
+    const effect = this.effects.get("protectiveMirage");
```

Comment:
Idle musing. We may someday want something like "effect hooks", instead of needing to treat this as a passive hook that is always evaluated and we check to see whether the relevant effect exists.

We could one day have a framework for effect hooks where only specifically identified active effects get their hooks called (otherwise working the same way as actor hooks do).